functions/HttpTriggerAPIUsers/__init__.py

```diff
@@ -1,52 +1,137 @@
-    logging.info('Python HTTP Database trigger function processed a request.')
-    
+    logging.info(
+        'Python HTTP trigger for /users function is processing a request.')
+
+    # Check request method
+    method = req.method
+    if not method:
+        logging.critical('No method available')
+        raise Exception('No method passed')
+
+    # Create a new connection
+    logging.debug("Attempting DB connection!")
+    try:
+        conn = get_db_connection()
+    except (pyodbc.DatabaseError, pyodbc.InterfaceError) as e:
+        logging.critical("Failed to connect to DB: " + e.args[0])
+        logging.debug("Error: " + e.args[1])
+        return func.HttpResponse(status_code=500)
+        
+    logging.debug("Connection to DB successful!")
+
+    try:
+        # Return results according to the method
+        if method == "GET":
+            logging.debug("Attempting to retrieve users...")
+            all_users_http_response = get_users(conn)
+            logging.debug("Users retrieved successfully!")
+            return all_users_http_response
+
+        elif method == "POST":
+            logging.debug("Attempting to add user...")
+            user_req_body = req.get_json()
+            new_user_id_http_response = add_user(conn, user_req_body)
+            logging.debug("User added successfully!")
+            return new_user_id_http_response
+
+        else:
+            logging.warn(f"Request with method {method} has been recieved, but that is not allowed for this endpoint")
+            return func.HttpResponse(status_code=405)
+
+    #displays erros encountered when API methods were called
+    except Exception as e:
+        return func.HttpResponse("Error: %s" % str(e), status_code=500)
+    finally: 
+        conn.close()
+        logging.debug('Connection to DB closed')
+
+
+def get_db_connection():
     # Database credentials.
     server = os.environ["ENV_DATABASE_SERVER"]
     database = os.environ["ENV_DATABASE_NAME"]
     username = os.environ["ENV_DATABASE_USERNAME"]
     password = os.environ["ENV_DATABASE_PASSWORD"]
-    
+
     # Define driver
     driver = '{ODBC Driver 17 for SQL Server}'
-    
+
     # Define the connection string
-    connection_string = "Driver={};Server={};Database={};Uid={};Pwd={};Encrypt=yes;TrustServerCertificate=yes;Connection Timeout=30;".format(driver, server, database, username, password)
-    
-    # Create a new connection
-    try:
-        with pypyodbc.connect(connection_string) as conn:
-            return select_users_query(conn)
-    except pypyodbc.DatabaseError as e:
-        if e.args[0] == '28000':
-          return func.HttpResponse(
-              "Unauthorized",
-              status_code=403
-          )
-
-# Define query    
-def select_users_query(conn):
+    connection_string = "Driver={};Server={};Database={};Uid={};Pwd={};Encrypt=yes;TrustServerCertificate=yes;Connection Timeout=30;".format(
+        driver, server, database, username, password)
+
```

Comment:
Potential for sql injection

Reply:
I don't know that that's true. These variables come from our environment variables so it's not really something a user has access to. Also, I'm not sure that ? would work because we aren't executing a statement.